### ISSUE TITLE
What's the use of tform.inverse() in transform.PiecewiseAffineTransform?

### ISSUE THREAD

**Issue by leeeeeeo** [scikit-image/scikit-image#3306]
## Description
In the [skimage.transform.PiecewiseAffineTransform example](http://scikit-image.org/docs/dev/auto_examples/transform/plot_piecewise_affine.html), blue points come from `tform.inverse()`, from this line:

`ax.plot(tform.inverse(src)[:, 0], tform.inverse(src)[:, 1], '.b')`

I draw destination points in red, It looks like this : 

![1533117427404](https://user-images.githubusercontent.com/25877709/43515139-5dbbcdb2-95b4-11e8-9919-19a4fef20f0f.jpg)

I have two questions about `skimage.transform.PiecewiseAffineTransform`.

1. Destination points (red points) and output image is not coherent. Suppose the image and inversed points are changed by `sin()` waves. But the destination points are changed by `1-sin()`. So what's the meaning of destination points?

2. What is `tform.inverse()`? I looked up, and annotation says it apply inverse transformation.
But Why the inverse transformation isn't between destination points(red points) and source points(source mesh)?

Thanks for all of you. Looking forward to your answer.

I'm using python 2.7.15, Anaconda, scikit-image 0.13.1, macOS 10.13.6.

**Comment by leeeeeeo** [comment]
In my view, mesh warp works like this (image from google.) : 
![har11_meshwarp_network-002](https://user-images.githubusercontent.com/25877709/43515554-50f1a240-95b5-11e8-9b75-e3cd302e3a8c.png)
People in the image should be warped follow the twisted mesh, which is the destination mesh. But the image in the [example](http://scikit-image.org/docs/dev/auto_examples/transform/plot_piecewise_affine.html) seems opposite from destination points.

**Comment by hmaarrfk** [comment]
Maybe @stefanv 's tutorial can help you. It helped me a few months back:

https://github.com/scikit-image/skimage-tutorials/blob/master/lectures/adv4_warping.ipynb

**Comment by emmanuelle** [comment]
Hi @leeeeeeo . The use of both the transform and its inverse comes from the following identity: if you have a transform ``f``, a source ``src`` and destination ``dest`` images, so that
```
f(src) = dest
```
then 
```
dest[x, y] = src[f^{-1}(x), f^{-1}(y)]
```
that is, the value of a pixel of ``dest`` is the value of ``src`` at  inverse transform of pixel coordinates. This is more easily understood if you take the example of a zoom factor (say, you want to blow up your image by a factor of 2, then for pixel values of the new image you look for pixel values of the src image at x/2, y/2).

**Comment by emmanuelle** [comment]
I hope it helps, the text of the example could surely be longer and explain more.

**Comment by zihaozhang9** [comment]
![image](https://user-images.githubusercontent.com/21007783/84012720-8fc7ec80-a9aa-11ea-817f-986e25e244a5.png)

It looks like this:
`dst_rows = src[:, 1]-np.sin(np.linspace(0, 3 * np.pi, src.shape[0])) * 50`
-sin is shown above.
But in the image, the coordinates of the upper left corner are (0,0). Therefore, the graph of dst (red dot) is opposite to that of -sin.

The second parameter of the [`warp` function](https://scikit-image.org/docs/stable/api/skimage.transform.html#warp) is inverse_map. Inverse coordinate map, which transforms coordinates in the output images into their corresponding coordinates in the input image.

![image](https://user-images.githubusercontent.com/21007783/84013651-ca7e5480-a9ab-11ea-9531-ad4a79dbca62.png)


`temp = - np.sin(np.linspace(0, 3 * np.pi, src.shape[0])) * 50` <br>
`ax.plot(dst_cols[0:], temp[0:], '.g')`

So apply tform to dst image to get src image.
`src=tform(dst)`

**Comment by stefanv** [comment]
With PiecewiseAffine, you will have to estimate the warp in the opposite direction, instead of using `inverse`.  We also provide

```
inverse_affines : list of AffineTransform objects
    Inverse affine transformations for each triangle in the mesh.
```

**Comment by stefanv** [comment]
I stand corrected, it looks as though someone *did* implement the inverse transform for PiecewiseAffine!

**Comment by zihaozhang9** [comment]
> I stand corrected, it looks as though someone _did_ implement the inverse transform for PiecewiseAffine!

I think` out = warp(image, tform)` is the inverse transform, and `out = warp(image, tform.inverse) `is the positive transform. Is my understanding correct?

**Comment by stefanv** [comment]
Yes, you have to provide the inverse transform for the forward warp because, in that process, you map the position of each output pixel *back* to the input image and interpolate to get its value.
